### PR TITLE
fix(javascript) incorrect function name highlighting

### LINKED ISSUE #3870: (JavaScript) Function names with a prefix shared with a built-in global are confused for a global
**Which language seems to have the issue?**
JavaScript and TypeScript

**Are you using `highlight` or `highlightAuto`?**
`highlight`

**Sample Code to Reproduce**

```js
eval();
neval();
eval2();
```
Screenshot 2023-09-22 at 07 25 08

**Expected behavior**

An identifier that starts with the prefix of a built-in global should not be recognized as that built in global. For example, the identifiers `eval2`, `evaluate`, `evaluation` are all confused with the built-in global `eval` and highlighted incorrectly.

The expected highlithing:

Screenshot 2023-09-22 at 07 25 49

### PR DESCRIPTION
**Pull request by cyfung1031**
* fix #3870
* override #3871

### Before

Screen Shot 2023-12-13 at 20 46 11

### After

Screen Shot 2023-12-13 at 20 46 08

cc @joshgoebel @arnog

### CODE REVIEW

File: src/languages/javascript.js [hunk 1]
```diff
@@ -592,7 +592,7 @@ export default function(hljs) {
         relevance: 0
       },
       {
-        match: [ /\bconstructor(?=\s*\()/ ],
+        match: [ new RegExp(`\\b(constructor|${ECMAScript.BUILT_IN_GLOBALS.map(x=>`${x}\\w+`).join('|')}|${ECMAScript.BUILT_IN_VARIABLES.map(x=>`${x}\\w+`).join('|')})(?=\\s*\\()`) ],
```
Comment (joshgoebel) [inline src/languages/javascript.js:595]: Why so limited though vs matching just any valid function name?

```
[IDENT]()
```

Comment (cyfung1031) [reply src/languages/javascript.js:595]: This is against ```js
  const FUNCTION_CALL = {
    match: regex.concat(
      /\b/,
      noneOf([
        ...ECMAScript.BUILT_IN_GLOBALS,
        "super",
        "import"
      ]),
      IDENT_RE, regex.lookahead(/\(/)),
    className: "title.function",
    relevance: 0
  };
  ```

Comment (joshgoebel) [reply src/languages/javascript.js:595]: Oh right, we already do this... so isn't this just a bug/problem with the `noneOf` guard?  This seems to fix it:

```js
noneOf([
        ...ECMAScript.BUILT_IN_GLOBALS,
        "super",
        "import"
      ].map(x => `${x}\\(`)),
```

Thoughts?

Comment (joshgoebel) [reply src/languages/javascript.js:595]: More complete to deal with spaces also:

```
  const FUNCTION_CALL = {
    match: regex.concat(
      /\b/,
      noneOf([
        ...ECMAScript.BUILT_IN_GLOBALS,
        "super",
        "import"
      ].map(x=> `${x}\\s*\\(`)),
      IDENT_RE, regex.lookahead(/\s*\(/)),
    className: "title.function",
    relevance: 0
  };
```

Comment (cyfung1031) [reply src/languages/javascript.js]: It looks good. I amended as what you suggested.

**Comment by joshgoebel** [comment]
Awesome, so now lets just add a changelog entry! :)

**Comment by cyfung1031** [comment]
> Awesome, so now lets just add a changelog entry! :)

done.

**Review by joshgoebel — APPROVED** [review]

**Comment by joshgoebel** [comment]
Will have to take another look later, tests aren't all passing.

**Comment by cyfung1031** [comment]
> Will have to take another look later, tests aren't all passing.

I am not quite sure what the error suggested.
Is it because of the change or the conflict with the latest commits?

btw, `regex.lookahead(/\(/))` also works. 
only `.map(x => `${x}\\s*\\(`))` is required.

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**